samples/azure/azure-service-fabric-routing/sample.md

```diff
@@ -0,0 +1,202 @@
+- Endpoint discriminators that are based on Service Fabric partition keys 
+- Mapping function per incoming message type that maps incoming messages to a partition key
+
+Use this code to enable Receiver-Side distribution:
+
+snippet: ConfigureReceiverSideDistribution-CandidateVoteCount
+
+In many low-throughput scenarios Receiver Side Distribution might be enough to get started. To achieve better performance, it would be more efficient to route directly to the partitions from the sender with *Sender-Side Distribution*.
+
+## Sender Side Distribution
+
+Receiver Side Distribution addresses forwarding messages that arrive to the incorrect partition. Forwarding received messages introduces some overhead though. To remove the overhead on the receiver side the Sender Side Distribution approach can be used to distribute messages to the correct endpoint instances based on Service Fabric partitioning information.
+
+Sender Side Distribution can be applied to endpoints hosted inside Service Fabric by using the partition information of the stateful services. This is suitable for endpoints hosted inside the cluster that need to send messages to other endpoints hosted in the cluster. For the endpoints hosted outside of the cluster access to Service Fabric APIs is not possible. Instead, partitioned destination endpoints need to be registered with message mapping for commands sent out.
+
+The Sender Side Distribution works in the following way:
+
+1. A mapping function is applied when dispatching messages. This mapping function is intended to select a partition key based on business criteria. In this example it's either the candidate name or the zip code of the voter, depending on the message type and destination endpoint.
+1. The result of this mapping, a partition key, is then added as a `partition-key` header to the outgoing message. This ensures that its value doesn't have to be calculated on the receiver side again and no receiver side distribution will occur. 
+1. With determined destination, the message is sent to the instance specific queue directly.
+
+### Partition aware distribution strategy
+
+The Sender Side Distribution feature adds a [custom distribution strategy](/nservicebus/messaging/routing-extensibility.md#routing-apis-message-distribution) `PartitionAwareDistributionStrategy` into the outgoing pipeline, which is responsible for selecting a destination queue for each message sent to a specific endpoint. When a destination is to be selected for a given outgoing message, the mapping function is applied to obtain the partition key value. The message has its `partition-key` header value set and the instance specific queue is selected as a destination address.
+
+### Configuration
+
+Sender Side Distribution is configured by providing partition information for a given endpoint and ensuring each of these partitions is uniquely addressable on the receiver side.
```

Comment:
I'm by far no English expert but we are talking about a location context? In this case point *at* the context?